src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -236,29 +235,27 @@
       (ca/get-certificate-statuses settings)
       (as-json-or-pson context))))
 
-(schema/defn routes
+(schema/defn ^:always-validate web-routes :- comidi/BidiRoute
   [ca-settings :- ca/CaSettings]
-  (compojure/context "/:environment" [environment]
-    (compojure/routes
-      (ANY "/certificate_status/:subject" [subject]
-        (certificate-status subject ca-settings))
-      (ANY "/certificate_statuses/:ignored-but-required" [do-not-use]
-        (certificate-statuses ca-settings)))
-      (GET "/certificate/:subject" [subject]
-        (handle-get-certificate subject ca-settings))
-      (compojure/context "/certificate_request/:subject" [subject]
-        (GET "/" []
-          (handle-get-certificate-request subject ca-settings))
-        (PUT "/" {body :body}
-          (handle-put-certificate-request! subject body ca-settings)))
-      (GET "/certificate_revocation_list/:ignored-node-name" []
-        (handle-get-certificate-revocation-list ca-settings))))
+  (comidi/context ["/" :environment]
+    (ANY ["/certificate_status/" :subject] [subject]
+         (certificate-status subject ca-settings))
+    (ANY ["/certificate_statuses/" :ignored-but-required] []
+         (certificate-statuses ca-settings))
+    (GET ["/certificate/" :subject] [subject]
+         (handle-get-certificate subject ca-settings))
+    (comidi/context ["/certificate_request/" :subject]
+      (GET [""] [subject]
+           (handle-get-certificate-request subject ca-settings))
+      (PUT [""] [subject :as {body :body}]
+           (handle-put-certificate-request! subject body ca-settings)))
+    (GET ["/certificate_revocation_list/" :ignored-node-name] []
+         (handle-get-certificate-revocation-list ca-settings))))
 
 (schema/defn ^:always-validate
-  build-ring-handler
-  [ca-settings :- ca/CaSettings
-   puppet-version :- schema/Str]
-  (-> (routes ca-settings)
-      ;(liberator-dev/wrap-trace :header)           ; very useful for debugging!
-      (ringutils/wrap-with-puppet-version-header puppet-version)
-      (ringutils/wrap-response-logging)))
+  wrap-middleware
+  [handler puppet-version]
```

Comment:
Useful to keep the schema types here since this is called from outside of this namespace?